go-controller/pkg/libovsdbops/switch.go

```diff
@@ -348,21 +349,22 @@ type logicalSwitchPortPredicate func(*nbdb.LogicalSwitchPort) bool
 func DeleteLogicalSwitchPortsWithPredicateOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, sw *nbdb.LogicalSwitch, p logicalSwitchPortPredicate) ([]libovsdb.Operation, error) {
 	sw, err := GetLogicalSwitch(nbClient, sw)
 	if err != nil {
-		return nil, fmt.Errorf("error getting switch %+v: %v", sw, err)
+		if errors.Is(err, libovsdbclient.ErrNotFound) {
+			return ops, nil
+		} else {
+			return nil, fmt.Errorf("error getting switch %+v: %v", sw, err)
+		}
 	}
 
-	lsps := []*nbdb.LogicalSwitchPort{}
+	var lsps []*nbdb.LogicalSwitchPort
 	for _, port := range sw.Ports {
 		lsp := &nbdb.LogicalSwitchPort{UUID: port}
-		lsp, err := GetLogicalSwitchPort(nbClient, lsp)
+		lsp, err = GetLogicalSwitchPort(nbClient, lsp)
 		if err != nil {
-			return nil, fmt.Errorf("error getting port %+v: %v", lsp, err)
+			return ops, fmt.Errorf("error getting port %+v: %v", lsp, err)
```

Comment:
why not check here if the error is also "object not found". If the LSP doesnt exist, then we can jsut return ops, nil right? Since the LSP is already deleted.

Reply:
The LS strongly references the LSP, it would be an integrity violation for an LSP referenced from a LS to not exist. 
Unsure if we would like to catch that or ignore it in the event it were to happen so I did not suggest anything :P

Reply:
+1 . One pathological case is if the lsp (with same name, like a pod from a stateful set) moved to a different node (i.e. different ls) and we end up in a out of sync situation. But yeah, lsp must have an ls in order to exist in the db.

Reply:
but these are two separate instances in time... Just because I did a call to get the ports on a switch, and then later I do another get call on the port doesn't mean it will still be there. I think we are all saying this should never happen, but it is still possible it could somehow happen. In that case we should ignore the error if it didnt exist because our goal here is to delete it and if it is gone, signaling and error and retrying wont do anything.

Reply:
I was thinking that it could not be other than us deleting the LSP so that would not be possible. But you are right, we don't even need to think about it and we should ignore the error.